raiden-ts/src/channels/epics.ts

```diff
@@ -480,10 +471,12 @@ function mapChannelEventsToAction(
         let action;
         switch (topic) {
           case openTopic: {
-            const [, p1, p2, settleTimeout] = args as ChannelOpenedEvent;
+            const [, p1, p2, settleTimeout] = args as EventTuple<
```

Comment:
Can we get rid of this exhaustive syntax and just declare them as before but have the proper types? So instead of

```ts
type ChannelOpenedEvent = [BigNumber, Address, Address, BigNumber, Event];
```

we have 

```ts
type TokenNetworkEvent<E extends string> = EventTuple<ContractFilter<TokenNetwork, E>;
type ChannelOpenedEvent = TokenNetworkEvent<'ChannelOpened'>;
```

Does that not work? Is it stupid? ...?

Reply:
Yes, something like that would work (preferably with a stricter constraint on the `E` generic, but that's the idea), but they were used only in **one** place each, so I thought using `EventTuple<ContractFilter<TokenNetwork, 'ChannelOpened'>>` directly was already descriptive enough. I'll do it like you suggested then, just to be more descriptive.

Reply:
> eferably with a stricter constraint on the E generic,

Yeah, probably then `E extends ChannelEventNames`?

> but they were used only in one place each, so I thought using 

At least one is used twice :stuck_out_tongue: 

> directly was already descriptive enough. I'll do it like you suggested then, just to be more descriptive.

Yeah, but unfortunately this always leads to line breaks which make a single simple line to become 3. And the expressions meaning itself is actually so simple, but reads complicated. So not as itself, but in the flow of a function it is quite a lot for a type cast. I think having one like of type declaration outside the function and then one short type-cast inside is even less numbers. And I would argue it reads better.
But please have your own arguments. Your PR is really good. Was the only thing I found. :stuck_out_tongue:

Reply:
With a default generic parameter, we can use `ChannelEvents` for that. 
:heavy_check_mark: Done.